demisto_sdk/commands/common/tools.py

```diff
@@ -2818,9 +2823,75 @@ def compare_context_path_in_yml_and_readme(yml_dict, readme_content):
     return different_contexts
 
 
-def write_yml(yml_path: str, yml_data: Dict):
-    with open(yml_path, "w") as f:
-        yaml.dump(yml_data, f)  # ruamel preservers multilines
+def safe_write_unicode_yml_or_json(
+    path: Union[Path, str],
+    data: Dict,
+    handler: XSOAR_Handler,
+    indent: int = 0,
+    sort_keys: bool = False,
+    **kwargs,
+):
+    """
+    Write unicode content into a json/yml file.
+    """
+    path = Path(path)
+
+    if isinstance(handler, YAML_Handler):
+        safe_write_unicode_yml(
+            path,
+            yml_data=data,
+            handler=handler,
+            indent=indent,
+            sort_keys=sort_keys,
+            **kwargs,
+        )
+    else:
+        safe_write_unicode_json(
+            path,
+            json_data=data,
+            handler=handler,
+            indent=indent,
+            sort_keys=sort_keys,
+            **kwargs,
+        )
+
+
+def safe_write_unicode_yml(
```

Comment:
what's the difference between this one and the `_json` one? They seem to do pretty much the same thing.

Reply:
1. one is for yml, while the other is for json
2. the default handler in each case is different, `yaml` for dumpring yml and `json` for dumping `json`

Even though they are very similar, I think we **should have** separate logic for each file type

Reply:
will use only `write_dict` for all